src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -39,13 +96,33 @@ public PlayerEditor() {
 
         add(new JScrollPane(playerInfoList,
                 JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_NEVER), "span, grow");
+
+        playerInfoList.addListSelectionListener(event -> SwingUtilities.invokeLater(() -> {
+            if (!event.getValueIsAdjusting()) {
+                List<Integer> listToRemove = new ArrayList<>();
+                for (int i = 0; i < playersModel.size() - main.config.PLAYER_INFOS.size(); i++) {
+                    if (!nearbyPlayerList.contains(playersModel.get(i + main.config.PLAYER_INFOS.size()))) {
+                        if (!playerInfoList.getSelectedValuesList().contains(playersModel.get(i + main.config.PLAYER_INFOS.size()))) {
+                            listToRemove.add(i + main.config.PLAYER_INFOS.size());
+                        }
+                    }
+                }
+                Collections.reverse(listToRemove);
+                for (int p : listToRemove) {
+                    playersModel.remove(p);
+                }
```

Comment:
This is one very convoluted way of removing elements. I think the list selection change listener has a start and end interval change, or something similar, you only need to iterate those elements, as long as they belong to the "nearby player list".

https://docs.oracle.com/javase/tutorial/uiswing/events/listselectionlistener.html